lib/src/rules/use_build_context_synchronously.dart

```diff
@@ -119,24 +119,33 @@ class _Visitor extends SimpleAstVisitor {
   }
 
   void check(AstNode node) {
+    bool checkStatements(AstNode child, NodeList<Statement> statements) {
+      var index = statements.indexOf(child as Statement);
+      for (var i = index - 1; i >= 0; i--) {
+        var s = statements.elementAt(i);
+        if (isMountedCheck(s)) {
+          return false;
+        } else if (isAsync(s)) {
+          rule.reportLint(node);
+          return true;
```

Comment:
If we've already reported a lint, is there any reason to keep looking up the flow path taken to get here? Could this produce multiple lints on the same node?

Reply:
Hmmm.  Interesting.  Will have to think on this.  Good question.